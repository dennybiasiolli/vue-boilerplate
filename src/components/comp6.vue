<template>
  <div>
    <p>{{ message }}</p>
    <v-text-field
      v-model="message"
      label="Label Text"
    />
  </div>
</template>

<script>
export default {
  data() {
    return {
      message: 'Hello Vue!'
    }
  }
}
</script>
