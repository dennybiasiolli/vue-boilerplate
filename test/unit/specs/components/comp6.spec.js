import { shallow } from '@vue/test-utils'
import comp6 from '@/components/comp6'

describe('comp6', () => {
  test('should match snapshot', () => {
    const component = shallow(comp6)
    expect(component.element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(typeof comp6.data).toBe('function')
    const defaultData = comp6.data()
    expect(defaultData.message).toBe('Hello Vue!')
  })

  // Inspect the component instance on mount
  test('correctly sets the seen flag when created', () => {
    const vm = shallow(comp6).vm
    expect(vm.message).toBe('Hello Vue!')
  })

  // Mount an instance and inspect the render output
  test('renders the correct message', () => {
    const vm = shallow(comp6).vm
    expect(vm.$el.textContent).toContain('Hello Vue!')
    expect(vm.$el.querySelector('p').textContent).toBe('Hello Vue!')
  })
})
