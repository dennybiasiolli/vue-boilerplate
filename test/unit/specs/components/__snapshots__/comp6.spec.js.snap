// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comp6 should match snapshot 1`] = `
<div>
  <p>
    Hello Vue!
  </p>
   
  <input />
</div>
`;
